Sources/GravatarUI/DesignSystem/UIFont+DesignSystem.swift

```diff
@@ -0,0 +1,46 @@
+import Foundation
+import UIKit
+
+extension UIFont {
+    /// Whether to adjust the font size according to the system's font size settings. Default: `true`.
+    public static var isGravatarDynamicFontSizeEnabled: Bool = true
+
+    // Point values noted here comes from the "Large (Default)" column of the
+    // Specification: https://developer.apple.com/design/human-interface-guidelines/typography#Specifications
+    // Note: These are dynamic sized fonts so the real size will vary.
+    enum DS {
+        static var largeTitle: UIFont { DynamicFontHelper.fontForTextStyle(.largeTitle, fontWeight: .bold) } // 34pt
+        static var title1: UIFont { DynamicFontHelper.fontForTextStyle(.title1, fontWeight: .bold) } // 28pt
+        static var title2: UIFont { DynamicFontHelper.fontForTextStyle(.title2, fontWeight: .bold) } // 22pt
+        static var title3: UIFont { DynamicFontHelper.fontForTextStyle(.title3, fontWeight: .semibold) } // 20pt
+
+        enum Body {
+            static var small: UIFont { DynamicFontHelper.fontForTextStyle(.subheadline, fontWeight: .regular) } // 15pt
+            static var medium: UIFont { DynamicFontHelper.fontForTextStyle(.callout, fontWeight: .regular) } // 16pt
+            static var large: UIFont { DynamicFontHelper.fontForTextStyle(.body, fontWeight: .regular) } // 17pt
+
+            enum Emphasized {
+                static var small: UIFont { DynamicFontHelper.fontForTextStyle(.subheadline, fontWeight: .semibold) } // 15pt
+                static var medium: UIFont { DynamicFontHelper.fontForTextStyle(.callout, fontWeight: .semibold) } // 16pt
+                static var large: UIFont { DynamicFontHelper.fontForTextStyle(.body, fontWeight: .semibold) } // 17pt
+            }
+        }
+
+        static var footnote: UIFont { DynamicFontHelper.fontForTextStyle(.footnote, fontWeight: .regular) } // 13pt
+        static var caption: UIFont { DynamicFontHelper.fontForTextStyle(.caption1, fontWeight: .regular) } // 12pt
+    }
+}
+
+private enum DynamicFontHelper {
+    static func fontForTextStyle(_ style: UIFont.TextStyle, fontWeight weight: UIFont.Weight) -> UIFont {
```

Comment:
Small detail: Apple APIs usually break the function name into name(label in cases like this:
```suggestion
    static func font(forTextStyle style: UIFont.TextStyle, weight: UIFont.Weight) -> UIFont {
```
It basically changes where the `(` is, bit it makes it nicer to split it in different lines when needed:

```swift
DynamicFontHelper.font(
    forTextStyle: .body,
    weight: .bold
)
```

We are also in the context of font creation (by the function name) so, I'd say that the font `fontWeight` is not really needed 🤔

Reply:
Yep, much better.